systems_cross_building/cross_building_v2.rst

```diff
@@ -0,0 +1,165 @@
+        app -> testtool [label="BR(host)"]
+        
+        cmake -> cmakelib -> zlibb
+        app -> cmake [label="BR"]
+    }
+
+
+You can find all these recipes in this [this repository](https://github.com/jgsogo/conan-xbuild),
+all the packages has the same structure, these are the relevant points:
+
+* Same set of settings, nothing different for executables, tools or libraries:
+
+  .. code-block:: python
+    
+     settings = "os", "arch", "compiler", "build_type"
+
+* The ``build()`` method add a message with the settings values to be used by the C++ code (very useful
+  to know the settings used to compile):
+
+  .. code-block:: python
+
+     def build(self):
+         cmake = CMake(self)
+         cmake.definitions["MESSAGE:STRING"] = "|".join(map(str, [self.settings.os, self.settings.arch, self.settings.compiler, self.settings.build_type]))
+         cmake.configure()
+         cmake.build()
+
+* Every package contains a library. ``app``, ``testtool`` and ``cmake`` also contains a executable. libraries
+  link to the libraries of the requirements. Executables link to the libraries in their package.
+
+* C++ implementatio is quite simple: there is function and an inline function in every library (useful to check shared builds),
+  both of them identifies the library and show the message. Functions call to their requirements too.
+
+  This is the output for the ``cmake_ese`` application compiled with a ``Debug`` profile, for example:
+
+  .. code-block:: bash
+
+     > cmake_exe: Macos|x86_64|apple-clang|Debug
+     > cmake_header: Macos|x86_64|apple-clang|Debug
+     > cmake: Macos|x86_64|apple-clang|Debug (shared!)
+         > cmakelib_header: Macos|x86_64|apple-clang|Debug
+         > cmakelib: Macos|x86_64|apple-clang|Debug (shared!)
+             > zlib_header: Macos|x86_64|apple-clang|Debug
+             > zlib: Macos|x86_64|apple-clang|Debug shared!
+
+  Note the cascade calls and the ``shared`` flag that is printed from the libraries.
+
+
+User interface
+--------------
+
+Command line
+++++++++++++
+
+In order to start using the new cross-building feature you need to provide a profile for the *build_machine* too, so every
+command now accepts the arguments ``--profile:host`` (``-pr:h``) and ``--profile:build`` (``-pr:b``). The old argument
+``--profile`` (``-p``) can still be used and it will be assigned to the profile for the *host machine*.
+
+Other command line options like ``--settings``, ``--options`` and ``--env`` have been unfolded as well and
+should be assigned to the matching profile.
+
+For example, given the recipes in the [repository mentioned above](https://github.com/jgsogo/conan-xbuild), after
+exporting the recipe, the following command will generate our ``app`` for the ``profile_host`` using some tools
+like ``cmake`` built with the ``profile_build``:
+
+.. code-block:: bash
+
+   conan conan install app/0.1@user/testing --build --profile:host=profiles/profile_host --profile:build=profiles/profile_build
```

Comment:
What about profile:target ? Like target profile

Reply:
It could use the same CLI if needed, but I have serious doubts if target should be a profile or just an option. Nevertheless I think that cross-compilers are out of the scope of this iteration, do you think it is needed from the very beginning?

Reply:
Oh forget about. I thought using "target" instead of "building", but the gnu triplet already has host-build-target. Keep it as is, that's good enough.